test/e2e/util.go

```diff
@@ -313,112 +310,3 @@ func getNodeAddresses(node *v1.Node) (string, string) {
 	}
 	return ipv4Res, ipv6Res
 }
-
-// This sets up a listener that replies with the hostname, both on tcp and on udp
-func setupHostnameServers(container, address string, udpPort, tcpPort int) error {
```

Comment:
would this be useful to leave in this file or in a library as other tests may want to use this?

Reply:
Right now, it serves only the "external gateway" tests, which are moved in a separate file. 
I feel it's natural for it to be there now, as we don't have a pkg structure here and it is located together with its client.

If a test needs it, it's likely to end up in this new file, but we can also move the function back if we see that is being consumed by multiple tests from different places.